psl-core/src/main/java/org/linqs/psl/evaluation/statistics/RankingEvaluator.java

```diff
@@ -21,255 +21,254 @@
 
-        // True positrive rate (TPR) is along the Y-axis.
-        double prevY = 0.0;
-        // False positive rate (FPR) is along the X-axis.
-        double prevX = 0.0;
+        Sum these reciprocal ranks and take the mean.
+        */
 
-        // Go through the atoms from highest truth value to lowest.
-        for (GroundAtom atom : predicted) {
-            Boolean label = getLabel(atom);
-            if (label == null) {
+        for(GroundAtom atom : truthAtoms) {
+            if (atom.getValue() < threshold) {
                 continue;
             }
 
-            // Assume we predicted everything positive.
-            if (label != null && label) {
-                tp++;
-            } else {
-                fp++;
-            }
+            rankedAtomCount++;
 
-            double newY = (double)tp / (double)totalPositives;
-            double newX = (double)fp / (double)totalNegatives;
+            Constant category = getArgAtPosition(atom, categoryIndex);
+            int rank = getAtomRank(atom, categoryMap.get(category));
 
-            area += 0.5 * (newX - prevX) * Math.abs(newY - prevY) + (newX - prevX) * newY;
-            prevY = newY;
-            prevX = newX;
-        }
+            // If a truth atom doesn't have a corresponding
+            // target atom, skip it
 
-        // Add the final piece.
-        area += 0.5 * (1.0 - prevX) * Math.abs(1.0 - prevY) + (1.0 - prevX) * 1.0;
+            if (rank == 0) {
+                continue;
+            }
 
-        return area;
+            rr += 1 / (double) rank;
+        }
+
+        categoryMap.clear();
+        return (double) rr / rankedAtomCount;
     }
 
     @Override
-    public String getAllStats() {
-        return String.format(
-                "AUROC: %f, Positive Class AUPRC: %f, Negative Class AUPRC: %f",
-                auroc(), positiveAUPRC(), negativeAUPRC());
+    public double getRepMetric() {
+        switch (representative) {
+            case MRR:
+                return mrr();
+            default:
+                throw new IllegalStateException("Unknown representative metric: " + representative);
+        }
     }
 
-    /**
-     * If the atom exists in the truth, return it's boolean value.
-     * Otherwise return null.
-     */
-    private Boolean getLabel(GroundAtom atom) {
-        int index = truth.indexOf(atom);
-        if (index == -1) {
-            return null;
+    @Override
+    public double getBestRepScore() {
+        switch (representative) {
+            case MRR:
+                return 1.0;
+            default:
+                throw new IllegalStateException("Unknown representative metric: " + representative);
         }
+    }
 
-        return truth.get(index).getValue() > threshold;
+    @Override
+    public boolean isHigherRepBetter() {
+        return true;
+    }
+
+    @Override
+    public String getAllStats() {
+        return String.format("MRR: %f", mrr());
     }
-}
+}
```

Comment:
Make sure to end files with a newline.

(Note that this does not mean have a blank line at the end of the file. All proper unix lines end with the newline character (including the last line). This is often an editor setting.)